### PR TITLE
Add GitHub Actions CI Job

### PR DESCRIPTION
**Pull request by jotompki**
*Issue #, if available:*

*Description of changes:* Adds CI job that achieves parity with the current travis PR run (runs pre-commit). The other file also handles releasing to PyPI. There is a scoped token that we have uploaded as a github secret that we can use to publish to our PyPI repo. Tested on my repo with a test token and test repository:
```
/usr/bin/docker run --name a9377fd141b0b7cce9fa0615d628_7d5844 --label 179394 --workdir /github/workspace --rm -e AWS_DEFAULT_REGION -e pythonLocation -e LD_LIBRARY_PATH -e INPUT_PASSWORD -e INPUT_REPOSITORY_URL -e INPUT_USER -e INPUT_PACKAGES_DIR -e INPUT_VERIFY_METADATA -e INPUT_SKIP_EXISTING -e INPUT_VERBOSE -e HOME -e GITHUB_JOB -e GITHUB_REF -e GITHUB_SHA -e GITHUB_REPOSITORY -e GITHUB_REPOSITORY_OWNER -e GITHUB_RUN_ID -e GITHUB_RUN_NUMBER -e GITHUB_RETENTION_DAYS -e GITHUB_ACTOR -e GITHUB_WORKFLOW -e GITHUB_HEAD_REF -e GITHUB_BASE_REF -e GITHUB_EVENT_NAME -e GITHUB_SERVER_URL -e GITHUB_API_URL -e GITHUB_GRAPHQL_URL -e GITHUB_WORKSPACE -e GITHUB_ACTION -e GITHUB_EVENT_PATH -e GITHUB_ACTION_REPOSITORY -e GITHUB_ACTION_REF -e GITHUB_PATH -e GITHUB_ENV -e RUNNER_OS -e RUNNER_TOOL_CACHE -e RUNNER_TEMP -e RUNNER_WORKSPACE -e ACTIONS_RUNTIME_URL -e ACTIONS_RUNTIME_TOKEN -e ACTIONS_CACHE_URL -e GITHUB_ACTIONS=true -e CI=true -v "/var/run/docker.sock":"/var/run/docker.sock" -v "/home/runner/work/_temp/_github_home":"/github/home" -v "/home/runner/work/_temp/_github_workflow":"/github/workflow" -v "/home/runner/work/_temp/_runner_file_commands":"/github/file_commands" -v "/home/runner/work/cloudformation-cli/cloudformation-cli":"/github/workspace" 179394:5265a9377fd141b0b7cce9fa0615d628  "__token__" "***" "https://test.pypi.org/legacy/" "dist" "true" "false" "false"
Checking dist/cloudformation_cli-0.2.1-py3-none-any.whl: PASSED
Checking dist/cloudformation-cli-0.2.1.tar.gz: PASSED
Uploading distributions to https://test.pypi.org/legacy/
Uploading cloudformation_cli-0.2.1-py3-none-any.whl

  0%|          | 0.00/96.2k [00:00<?, ?B/s]
100%|██████████| 96.2k/96.2k [00:01<00:00, 66.1kB/s]
Uploading cloudformation-cli-0.2.1.tar.gz

  0%|          | 0.00/74.8k [00:00<?, ?B/s]
100%|██████████| 74.8k/74.8k [00:01<00:00, 69.1kB/s]

View at:
https://test.pypi.org/project/cloudformation-cli/0.2.1/

```

Very similar to https://github.com/aws-cloudformation/cloudformation-cli-python-plugin/pull/138

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/workflows/pr-ci.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  pull_request:
+    branches: [ master ]
+  release:
+    types: [ published ]
+
+jobs:
+  build:
+    env:
+      AWS_DEFAULT_REGION: us-east-1
+    runs-on: ubuntu-latest
```
Comment (PatMyron) [inline .github/workflows/pr-ci.yaml:17]: we should test on `macos-latest` and `windows-latest` as well

Comment (jotompki) [reply .github/workflows/pr-ci.yaml:15]: agree although was just shooting for parity with the current job in this PR. can see how easy would be to use  in addition. are we trying to do 9 jobs in total? os X python versions?

Comment (jotompki) [reply .github/workflows/pr-ci.yaml]: Windows [fails hard](https://github.com/aws-cloudformation/cloudformation-cli/pull/654/checks?check_run_id=1567545887). 

Are you OK with creating issues to follow up for windows? I'll try without

Comment (jotompki) [reply .github/workflows/pr-ci.yaml]: Got it fixed. just had to remove attempting to upgrade pip

**Comment by PatMyron** [comment]
Can we remove `.travis.yml` in these PRs as well? Easier to review the diff for parity if that's part of the PR

**Comment by jotompki** [comment]
> Can we remove [`.travis.yml`](https://github.com/aws-cloudformation/cloudformation-cli/blob/master/.travis.yml) in these PRs as well? Easier to review the diff for parity if that's part of the PR

Removed it in the last commit. Checks will show as incomplete since we require a travis success in the repo settings and there isn't one in this PR now. Will remove requirement once this is approved.

**Review by ammokhov — APPROVED** [review]

File: .github/workflows/pr-ci.yaml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  build:
+    env:
+      AWS_DEFAULT_REGION: us-east-1
```
Comment (PatMyron) [inline .github/workflows/pr-ci.yaml:14]: is this used for anything btw?
```suggestion
```

**Pull request merged**